### PR TITLE
Import PropTypes from 'prop-types'

### LINKED ISSUE #92: React.createClass + React.PropTypes warnings
With the [15.5.0 release](https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html), React is logging warnings about using the two items mentioned in the title. There is a [react-codemod](https://github.com/reactjs/react-codemod) available to migrate this -- but wasn't sure if you had anything else going on in the codebase dependent on older React versions.  Thanks!

### PR DESCRIPTION
**Pull request by kylecesmat**
Fixes #92 

This PR adds support for React 15.5+ by including the `prop-types` package. This also upgrades react-codemirror & offers a few housekeeping items.

- Upgrade `react-codemirror@1.0.0`
- Include `prop-types` package as a dependency
- Remove support for React < 15.4
- Use yarn for dependency management
- Upgrade travis configuration, remove old node versions

### CODE REVIEW

**Comment by ryan-roemer** [comment]
@kylecesmat -- If you're switching to yarn, you should update travis config too. Here's a (untested) suggestion:

```yaml
language: node_js

node_js:
  - "4"
  - "6"

sudo: false

branches:
  only:
    - master

cache: yarn

before_install:
  # GUI for real browsers.
  - export DISPLAY=:99.0
  - sh -e /etc/init.d/xvfb start

script:
  - yarn run check-ci

  # Prune deps to just production and ensure we can still build
  - yarn install --production --ignore-scripts --prefer-offline
  - yarn run build
```

**Comment by kylecesmat** [comment]
@ryan-roemer thanks for the suggestion! I modified the travis configuration, and removed the commands we should no longer need, now that we're using yarn.

Would you be able to provide a review?

**Comment by ryan-roemer** [comment]
@kylecesmat -- Restarting failed travis job on node6: https://travis-ci.org/FormidableLabs/component-playground/jobs/233708066

**Comment by kylecesmat** [comment]
@ryan-roemer yeah I'm seeing this error locally now on `node6`, but runs fine with `node4`. I'm assuming one of our phantomjs modules needs to be updated - I'm timeboxed on this right now but will investigate later.

**Comment by ryan-roemer** [comment]
@kylecesmat -- Feel free to comment out `- "6"` and place a ticket link there and ticket out "figure out phantom / update"

File: .travis.yml [hunk 1]
```diff
@@ -11,16 +12,14 @@ branches:
   only:
     - master
 
+cache: yarn
+
 before_install:
   # GUI for real browsers.
-  - npm install -g npm@3
```
Comment (chrisbolin) [inline .travis.yml]: How is yarn installed on Travis with this setup?

**Pull request merged**